### PR TITLE
ServiceAccounts: Add access control metadata to service accounts

### PR DESCRIPTION
**Pull request by Jguer**
**What this PR does / why we need it**:

- Add access control metadata to service accounts
  - Allows implementing role picker and permissions in UI
  
- Add avatars to service accounts

image

### CODE REVIEW

**Review by gamab — APPROVED** [review]
I can approve the metadata part and the avatar part. :rocket: 
For the Role part I'm missing context as to why it wasn't done before.

File: pkg/services/serviceaccounts/api/api.go [hunk 1]
```diff
@@ -132,14 +136,42 @@ func (api *ServiceAccountsAPI) ConvertToServiceAccount(ctx *models.ReqContext) r
...
+		return nil, nil
+	}
+
+	userPermissions, err := api.accesscontrol.GetUserPermissions(c.Req.Context(), c.SignedInUser)
+	if err != nil || len(userPermissions) == 0 {
+		api.log.Warn("could not fetch accesscontrol metadata for teams", "error", err)
+		return nil, err
+	}
+
+	return accesscontrol.GetResourcesMetadata(c.Req.Context(), userPermissions, "serviceaccounts", saIDs), nil
```
Comment (eleijonmarck) [inline pkg/services/serviceaccounts/api/api.go:172]: I was under the impression that this is a user can "create" , "edit" or "delete" a specific "scoped" resource.

But this is maybe something different

File: pkg/services/serviceaccounts/database/database.go [hunk 2]
```diff
@@ -143,25 +143,28 @@ func (s *ServiceAccountsStoreImpl) ListServiceAccounts(ctx context.Context, orgI
...
 		}
 		tokens, err := s.ListTokens(ctx, user.OrgId, user.UserId)
 		if err != nil {
 			return nil, err
 		}
 		saDTOs[i].Tokens = int64(len(tokens))
 	}
-	return saDTOs, err
+
+	return saDTOs, nil
```
Comment (eleijonmarck) [inline pkg/services/serviceaccounts/database/database.go:167]: love your space in code

**Comment by eleijonmarck** [comment]
> I can approve the metadata part and the avatar part. 🚀
> For the Role part I'm missing context as to why it wasn't done before.

@gamab we wanted to introduce service accounts incrementally, otherwise we would maybe bring in data that we actually do not need


Sort of like working with the thing you need to implement and grab the data you need at that time

**Pull request merged**